yt_dlp/extractor/funker530.py

```diff
@@ -0,0 +1,75 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+
+from .common import InfoExtractor
+from .rumble import RumbleEmbedIE
+from .youtube import YoutubeIE
+from ..utils import clean_html
+
+
+class Funker530IE(InfoExtractor):
+    _VALID_URL = r'https?:\/\/(?:www\.)?funker530\.com\/video\/(?P<id>[^\/]+)\/?'
+    # TODO: `python test/test_download.py TestDownload.test_Funker530` only runs the first test?
+    _TESTS = [{
+        'url': 'https://funker530.com/video/azov-patrol-caught-in-open-under-automatic-grenade-launcher-fire/',
+        'md5': '085f50fea27523a388bbc22e123e09c8',
+        'info_dict': {
+            'id': 'v2qbmu4',
+            'ext': 'mp4',
+            'title': 'Azov Patrol Caught In Open Under Automatic Grenade Launcher Fire',
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'uploader': 'Funker530',
+            'channel': 'Funker530',
+            'channel_url': 'https://rumble.com/c/c-1199543',
+            'width': 1280,
+            'height': 720,
+            'fps': 25,
+            'duration': 27,
+            'upload_date': '20230608',
+            'timestamp': 1686241321,
+            'live_status': 'not_live',
+            'description': 'md5:bea2e1f458095414e04b5ac189c2f980',
+        }
+    }, {
+        'url': 'https://funker530.com/video/my-friends-joined-the-russians-civdiv/',
+        'md5': '',
+        'info_dict': {
+            'id': '',
+            'ext': 'mp4',
+            'title': 'My “Friends” Joined the Russians - CivDiv',
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'uploader': '',
+            'width': '',
+            'height': '',
+            'fps': '',
+            'duration': '',
+            'upload_date': '',
+            'description': 'md5:',
+        }
+    }]
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+        webpage = self._download_webpage(url, display_id)
+        rumble = RumbleEmbedIE(downloader=self._downloader)
+        youtube = YoutubeIE(downloader=self._downloader)
+        found_rumble_videos = rumble._extract_embed_urls(url, webpage)
+        found_yt_videos = list(youtube._extract_embed_urls(url, webpage))  # TODO: is converting it to a list OK if there's only one video?
+        if len(list(found_yt_videos)):
+            embedded_video = youtube.extract(found_yt_videos[0])  # TODO: do I need to do all found videos or is [0] ok?
+        elif len(found_rumble_videos):
+            embedded_video = rumble.extract(found_rumble_videos[0])  # TODO: see above
```

Comment:
You were on the right track here. But since this site only offers embeds, IMO it makes more sense to return url results to their respective extractors. `_extract_embed_urls` is a classmethod so we can call it from the class.

If no videos are found on the page, you could do one of the following:
  1. `raise ExtractorError`: e.g. if something is wrong and we want a bug report about it
  2. `self.raise_no_formats`: e.g. other extraction steps are working but no formats can be found
  3. or, if metadata has been extracted and there isn't anything fatal or unbound before the return, you can let the extractor return a format-less info_dict and core code will `raise_no_formats`
```suggestion
        info = {}
        rumble_urls = RumbleEmbedIE._extract_embed_urls(url, webpage)
        if rumble_urls:
            info.update({'url': rumble_urls[0], 'ie_key': RumbleEmbedIE.ie_key()})
        else:
            youtube_url = next(YoutubeIE._extract_embed_urls(url, webpage), None)
            if youtube_url:
                info.update({'url': youtube_url, 'ie_key': YoutubeIE.ie_key()})
        if not info:
            raise ExtractorError('No videos found on webpage')
```